src/main/java/duke/commands/AddCommand.java

```diff
@@ -0,0 +1,69 @@
+package main.java.duke.commands;
+import java.io.IOException;
+
+import main.java.duke.*;
+import main.java.duke.tasks.Deadline;
+import main.java.duke.tasks.Event;
+import main.java.duke.tasks.Task;
+import main.java.duke.tasks.Todo;
+
+public class AddCommand extends Command {
+    private Task task;
+
+    /**
+     * Constructs a new add task command with the given task.
+     *
+     * @param task the given task
+     */
+    public AddCommand(Task task) {
+        this.task = task;
+    }
+
+    /**
+     * Executes the add task command.
+     *
+     * @param tasks given list of tasks
+     * @param gui given gui object
+     * @param storage given storage object
+     * @throws IOException
+     * @throws DukeException
+     */
+    public String execute(TaskList tasks, MainWindow gui, Storage storage) throws IOException, DukeException {
+        if (this.task instanceof Todo) {
+            storage.saveTaskToFile(task.toString() + "\n");
+            return addTask(this.task, tasks);
+        } else if (this.task instanceof Deadline) {
+            Deadline deadline = (Deadline) this.task;
```

Comment:
Is this casting needed? this.task remains as type task, and when it is passed to addTask() it is treated of type Task.